WordPress/Classes/ViewRelated/Media/MediaCellProgressView.swift

```diff
@@ -6,59 +6,78 @@ import WordPressShared
 /// asset is currently being processed, uploaded, or has failed to upload.
 ///
 class MediaCellProgressView: UIView {
-    let progressIndicator = ProgressIndicatorView()
-
-    private let retryContainer = UIStackView()
 
     enum State: Equatable {
         case stopped
         case retry
         case indeterminate
         case progress(Double)
+        case error
 
         static func ==(lhs: State, rhs: State) -> Bool {
             switch (lhs, rhs) {
             case (.stopped, .stopped): return true
             case (.retry, .retry): return true
             case (.indeterminate, .indeterminate): return true
             case let (.progress(l), .progress(r)): return l == r
+            case (.error, .error): return true
             default: return false
             }
         }
     }
 
+    // MARK: - public fields
+
+    let retryView = RetryView()
+    var errorTintColor = UIColor.white
+
     var state: State = .stopped {
         didSet {
-            switch state {
-            case .stopped:
-                progressIndicator.state = .stopped
-                retryContainer.isHidden = true
-            case .retry:
-                progressIndicator.state = .stopped
-                retryContainer.isHidden = false
-            case .indeterminate:
-                progressIndicator.state = .indeterminate
-                retryContainer.isHidden = true
-            case .progress:
-                progressIndicator.state = state
-                retryContainer.isHidden = true
-            }
+            refreshState()
         }
     }
 
-    override init(frame: CGRect) {
-        super.init(frame: frame)
+    var loaderAppearance: ProgressIndicatorView.Appearance {
```

Comment:
I'd perhaps rename this to `progressIndicatorAppearance` or something, just as we don't use the term 'loader' elsewhere in this class.